src/interfaces/uproxy_core_api.ts

```diff
@@ -215,14 +217,29 @@ export interface InvitationData {
 
 export enum PortControlSupport {PENDING, TRUE, FALSE};
 
-// Argument to #cloudInstall.
-export interface CloudInstallArgs {
-  // Use this cloud computing provider to create a server.
-  providerName: string;
-  // Provider-specific region in which to locate the new server.
-  region: string;
+export enum CloudOperationType {
+  CLOUD_INSTALL = 0,
+  CLOUD_DESTROY = 1
+}
+
+// Arguments to cloudUpdate
+export interface CloudOperationArgs {
+  operation: CloudOperationType;
+  // Use this cloud computing provider to access a server.
+  providerName :string;
+  // Provider-specific region in which to locate a new server.
+  region ?:string;
```

Comment:
BTW, I have to say my recommendation here, given the overhead required to add calls to the core, has to be just one cloud method with an "operation type" enum and some optional fields for some of the operations.